File: binary_search.py
def binary_search(list, item):
	low = 0
	high = len(list)-1
	
	while low <=high:
		mid = (low + high) / 2
		print("mid", mid)
		guess = list[mid]
		
		if guess == item:
			return "You will find this in the index" + str(mid)
		if guess > item:
			high =  mid - 1
		else:
			low = mid + 1
	return None;

my_list = [1, 3, 5, 7, 9]

#print binary_search(my_list, 2)
print binary_search(my_list, 9)

File: binary_search.js
function binary_search (list, item) {
	let low = 0;
	let high = list.length -1
	console.log("high", high)

	while (low <= high) {
		let mid = Math.floor((low + high) /2)
		let guess = list[mid]


		if (guess === item) {
			console.log("found it!!! Guess:" + guess + " Index:" + mid)
			return item
		} else if (guess < item){
			low = mid + 1
		} else if (guess > item){
			high = mid -1
		}
		//return "Item not found"
	}
}

console.log(binary_search([1, 2, 3, 4,], 3))
// console.log(binary_search([1, 2, 3, 4, 5], 7))
